yt_geotiff/io.py

```diff
@@ -127,6 +127,147 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
     
         return rv
 
+class rasterio_group(IOHandlerGeoTiff):
+    _dataset_type = "RasterioGroup"
+
+    # copy _read_fluid_selection
+    # Difference geotiff read specific band number
+    # 
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+
+        rv = {}
+        chunks = list(chunks)
+
+        if isinstance(selector, GridSelector):            
+
+            if not (len(chunks) == len(chunks[0].objs) == 1):
+                raise RuntimeError
+            g = chunks[0].objs[0]
+
+            if g.id in self._cached_fields:
+                gf = self._cached_fields[g.id]
+                rv.update(gf)
+
+            if len(rv) == len(fields):
+                return rv
+            
+            for field in fields:
+                if field in rv:
+                    self._hits += 1
+                    continue
+                self._misses += 1
+                ftype, fname = field 
+                
+                filename=self.ds.directory+'/'+self.ds._field_filename[fname]['filename']                              
```

Comment:
I mentioned encountering an issue where things don't work if the files are in different directories. I think you'll need to get rid of this line and if there is a place elsewhere where the directory is stripped off from the filename, you may need to restore that.

Reply:
We now read the full file path into the ds._field_filename attribute. In io.py, for the filename the full file path is called from the _field_filename dictionary.